queries/jsonnet/highlights.scm

```diff
@@ -30,63 +65,100 @@
   ","
   ";"
   ":"
-  "::"
-  ":::"
 ] @punctuation.delimiter
 
-; Nested field inheritance
-("+" @punctuation.delimiter [":" "::" ":::"])
-
-"=" @operator
-(unaryop) @operator
-(binary
-  operator: (_) @operator)
-
-"for" @repeat
-
-"function" @keyword.function
+[
+  "::"
+  ":::"
+] @punctuation.special
 
-"in" @keyword.operator
+(field
+  (fieldname) "+" @punctuation.special)
 
+; Imports
 [
- (local)
- (tailstrict)
- "assert"
-] @keyword
+  (import)
+  (importstr)
+] @include
 
-[
-  "else"
-  "if"
-  "then"
-] @conditional
+; References
 
-"error" @exception
+; Make reference same color as parameter 
+; (may incur performance issues on big files)
+; Depends on locals.scm
+((id) @parameter.reference
+ (#is? @parameter.reference parameter))
 
-[
-  (dollar)
-  (self)
-  (super)
-] @variable.builtin
-((id) @variable.builtin
- (#eq? @variable.builtin "std"))
+((id) @function.reference
+ (#is? @function.reference function))
+
+;((id) @var.reference
+; (#is? @var.reference var))
+;((id) @define
+; (#is? @var.reference var))
 
-; Function declaration
+; References do not apply to static field IDs
+; Workaround for `(#is-not? local)` not supported
+(fieldname (id) @field)
+(fieldname (string
+             (string_start) @text.strong
+             (string_content) @field
+             (string_end) @text.strong
+           ))
+
+; But it does apply if ID in an expression
+(fieldname
+ ("["
+  (id) @parameter.reference
+  "]"
+  (#is? @parameter.reference parameter)))
+;(fieldname
+; ("["
+;  (id) @define
+;  "]"
+;  (#is? @var.reference var)))
+
+; Functions
+(field
+  function: (fieldname (id) @function))
+(field
+  function: (fieldname
+              (string
+                (string_start) @text.strong
+                (string_content) @function
+                (string_end) @text.strong
+              )))
 (param
   identifier: (id) @parameter)
-(field
-  (fieldname (id) @function)
-  [(anonymous_function) (params)])
-(bind
-  function: (id) @function)
+
+(bind (id) @define)
+(bind function: (id) @function)
 
 ; Function call
-(functioncall (id) @function.call)
 (functioncall
   (fieldaccess
-    last: (id) @function.call))
-(functioncall
+    last: (id) @function.call
+  )?
   (fieldaccess_super
-    (id) @function.call))
-(named_argument (id) @parameter)
+    (id) @function.call
+  )?
+  (id)? @function.call
+  "("
+  (args
+    (named_argument
+      (id) @parameter
+    )
+  )?
+  ")"
+)
+
+; Emphasize implicit plus usage
+(implicit_plus
+  (_ "}"? @text.danger)
+  (object
+    "{" @text.danger
+  )
+)
```

Comment:
what even is this?

Reply:
There are two ways to combine expressions:

A binaryop:
```
expr + { }
```

And an implicit way:

```
expr { }
```

This leads to the same result but from the AST perspective this is not a binaryop but a separate concept (with lack of a better name I called it `implicit_plus` in the parser.

As the implicit behavior should be discouraged, I've included this piece to mark those curly brackets so it becomes apparent something implicit is happening here and people should be aware.

Reply:
Can you point out what's wrong about it?

Reply:
I mean just because it's discouraged by the community doesn't mean text.danger fits it at all, maybe punctuation.special to differentiate it from other brackets

Reply:
That doesn't feel expressive enough for me, if y'all don't want this, I'll rip it out and drop it in a plugin (with `after/`).

Reply:
yeah, of course you yourself can highlight stuff however you like - but as a sane default I'd say remove it vs using text.danger

Reply:
Treesitter is _not_ a linter: our goal is to correctly highlight _correct_ syntax; highlighting _incorrect_ syntax is a non-goal (we have LSP for that). Please remove this.

Reply:
It's gone.